chain.go

```diff
@@ -396,19 +397,23 @@ type FutureGetCoinSupplyResult chan *response
 
 // Receive waits for the response promised by the future and returns the
 // current coin supply
-func (r FutureGetCoinSupplyResult) Receive() (int64, error) {
+func (r FutureGetCoinSupplyResult) Receive() (*dcrutil.Amount, error) {
 	res, err := receiveFuture(r)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	// Unmarshal the result
-	var cs int64
+	var cs float64
```

Comment:
This `getcoinsupply` RPC does not return a value in Decred like many other RPCs, hence it should not be `float64`.  Rather it returns the value in atoms, and thus should be an `int64`.